extensions/data-transfer/portability-data-transfer-koofr/src/main/java/org/datatransferproject/transfer/koofr/photos/KoofrPhotosImporter.java

```diff
@@ -132,61 +141,74 @@ private String createAlbumFolder(PhotoAlbum album, KoofrClient koofrClient)
+        String parentPath = idempotentImportExecutor.getCachedValue(photo.getAlbumId());
+        String fullPath = parentPath + "/" + title;
 
-      String response;
+        if (koofrClient.fileExists(fullPath)) {
+          monitor.debug(() -> String.format("Photo already exists %s", photo.getTitle()));
 
-      try {
-        response = koofrClient.uploadFile(
-                parentPath, title, inMemoryInputStream, photo.getMediaType(), dateCreated, description);
-      } catch (KoofrClientIOException e) {
-        if (e.getCode() == 404) {
-          monitor.info(() -> String.format("Can't find album during importSingleItem for id: %s", photo.getDataId()), e);
-          response = "skipped-"+photo.getDataId();
+          return ItemImportResult.success(fullPath);
         }
-        else {
-          throw e;
+
+        final ByteArrayInputStream inMemoryInputStream = new ByteArrayInputStream(bytes);
+
+        try {
+          long inputStreamBytes = inputStreamWrapper.getBytes();
+          String stringResult = koofrClient.uploadFile(
+                  parentPath, title, inMemoryInputStream, photo.getMediaType(), dateCreated, description);
+          if (stringResult != null && !stringResult.isEmpty()) {
+            response = ItemImportResult.success(
+                    stringResult,
+                    inputStreamBytes);
+          } else {
+            response = ItemImportResult.success(String.format(SKIPPED_FILE_RESULT_FORMAT, photo.getDataId()));
+          }
+          size = inputStreamBytes;
+        } catch (KoofrClientIOException exception) {
+          if (exception.getCode() == 404) {
+            monitor.info(() -> String.format("Can't find album during importSingleItem for id: %s", photo.getDataId()), exception);
+            response = ItemImportResult.success(String.format(SKIPPED_FILE_RESULT_FORMAT, photo.getDataId()));
+          } else {
+            return ItemImportResult.error(exception, size);
+          }
         }
-      }
 
-      try {
-        if (photo.isInTempStore()) {
-          jobStore.removeData(jobId, photo.getFetchableUrl());
+        try {
+          if (photo.isInTempStore()) {
+            jobStore.removeData(jobId, photo.getFetchableUrl());
+          }
+        } catch (Exception e) {
+          // Swallow the exception caused by Remove data so that existing flows continue
+          monitor.info(
+                  () -> format("Exception swallowed while removing data for jobId %s, localPath %s",
+                          jobId, photo.getFetchableUrl()), e);
         }
-      } catch (Exception e) {
-        // Swallow the exception caused by Remove data so that existing flows continue
-        monitor.info(
-                () -> format("Exception swallowed while removing data for jobId %s, localPath %s",
-                        jobId, photo.getFetchableUrl()), e);
       }
 
       return response;
+    } catch (KoofrClientIOException exception) {
```

Comment:
I don't think we'd have any `KoofrClientIOException` here to be honest, as they are all caught in the upstream try/catch block.